Patched MARS Makefile does not correctly fetch Mars4_5.jar

For some reason running `make` does not fetch the source dependency `Mars4_5.jar`. Run using `GNU Make 3.81`:

```
$ ls
Makefile            README.md           java                mars-sockets        socket-examples.asm
20:59:20 thomasrussellmurphy@kaukauna:~/Documents/CWRU_2015a/EECS 314/Project/yams
$ make mars
mkdir -p mars-sockets
cd mars-sockets; jar xf ../Mars4_5.jar
java.io.FileNotFoundException: ../Mars4_5.jar (No such file or directory)
    at java.io.FileInputStream.open0(Native Method)
    at java.io.FileInputStream.open(FileInputStream.java:195)
    at java.io.FileInputStream.<init>(FileInputStream.java:138)
    at java.io.FileInputStream.<init>(FileInputStream.java:93)
    at sun.tools.jar.Main.run(Main.java:305)
    at sun.tools.jar.Main.main(Main.java:1233)
make: *** [mars-sockets/CreateMarsJar.bat] Error 1
```

'GNU Make 4.1` experiences the same error.

--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,7 @@ $(VANILLA_JAR):
 	wget $(MARS_URL)
 
 # Rule for extracting the JAR.
-$(MARS_DIR)/CreateMarsJar.bat:
+$(MARS_DIR)/CreateMarsJar.bat: $(VANILLA_JAR)
 	mkdir -p $(MARS_DIR)
 	cd $(MARS_DIR); jar xf ../$(VANILLA_JAR)
 

